File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        int idx = 0; //act like stack pointer
        for(char ch : s){
            if(idx>0 && 
            ((s[idx - 1] == 'A' && ch == 'B') || 
            (s[idx - 1] == 'C' && ch == 'D'))) {
                idx--;
            } else {
                s[idx++] = ch;
            }
        }
        return idx;
    }
};

File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.py
class Solution(object):
    def minLength(self, s):
        """
        :type s: str
        :rtype: int
        """
        stack = []
        for ch in s:
            if stack and stack[-1] + ch in ('AB', 'CD'):
                stack.pop()
            else:
                stack.append(ch)
        return len(stack)
        